src/python/pants/backend/docker/target_types.py

```diff
@@ -0,0 +1,42 @@
+# Copyright 2021 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from pants.engine.target import COMMON_TARGET_FIELDS, Dependencies, Sources, StringField, Target
+
+
+class DockerImageSources(Sources):
+    default = ("Dockerfile",)
+    expected_num_files = 1
+    help = "Name of the Dockerfile to use when building the Docker image."
+
+
+class DockerImageVersion(StringField):
+    alias = "version"
+    default = "latest"
+    help = "Image tag to apply to built images."
```

Comment:
Nice. A cool future development could be to let users set this dynamically via a plugin.